### ISSUE TITLE
[Bug] yarn failed to handle merged yarn.lock with duplicated key

### ISSUE THREAD

**Issue by missing1984** [yarnpkg/berry#1771]
- [*] I'd be willing to implement a fix

**Describe the bug**

Yarn failed to parse `yarn.lock` that includes duplicate keys, which can occur during git merge/rebase

**To Reproduce**

yarn.lock
```
# This file is generated by running "yarn install" inside your project.
# Manual changes might be lost - proceed with caution!

__metadata:
  version: 4

"babel-plugin-apply-mdx-type-prop@npm:1.6.16":
  version: 1.6.16
  resolution: "babel-plugin-apply-mdx-type-prop@npm:1.6.16"
  dependencies:
    "@babel/helper-plugin-utils": 7.10.4
    "@mdx-js/util": 1.6.16
  peerDependencies:
    "@babel/core": ^7.10.5
  checksum: 2/9c6132235351ef37a35cf97634b594efc6565327d9a7ca8a0975e3e0391dd70139c6d6f0e1d93480fef20adba204ce1d81a468946839c3690a3543c4c7beafb8
  languageName: node
  linkType: hard

"babel-plugin-apply-mdx-type-prop@npm:1.6.16":
  version: 1.6.16
  resolution: "babel-plugin-apply-mdx-type-prop@npm:1.6.16"
  dependencies:
    "@babel/helper-plugin-utils": 7.10.4
    "@mdx-js/util": 1.6.16
  peerDependencies:
    "@babel/core": ^7.10.5
  checksum: 2/9c6132235351ef37a35cf97634b594efc6565327d9a7ca8a0975e3e0391dd70139c6d6f0e1d93480fef20adba204ce1d81a468946839c3690a3543c4c7beafb8
  languageName: node
  linkType: hard
```

test.js

```
const fs = require("fs");
const { parseSyml } = require("@yarnpkg/parsers");
parseSyml(fs.readFileSync("./lock/yarn.lock", "utf-8"));
```
will raise error
```
YAMLException: duplicated mapping key at line 19, column -436:
    "babel-plugin-apply-mdx-type-pro ... 
**Screenshots**
```

**Environment if relevant (please complete the following information):**

 - OS: OSX
 - Node version 12
 - Yarn version 2.1.1

**Additional context**

`js-yaml` does not support custom duplicate handling, https://github.com/nodeca/js-yaml/issues/348

Let me know what would be the best way to fix this.

**Comment by paul-soporan** [comment]
🤔 How did you manage to get a lockfile with duplicate keys in the first place? I've checked and I don't think it's possible for our merge conflict auto-fixing to produce such results. If it's possible, can you write a repro for how you obtain duplicate keys?

**Comment by paul-soporan** [comment]
Sorry for not giving you an update on this, the update is [this comment](https://github.com/yarnpkg/berry/issues/1463#issuecomment-687302655) in another issue (it's about my project that would allow us to have a custom duplicate handler. We're currently considering if using it would be worth the bundle size increase cost).

**Comment by arcanis** [comment]
Ref: https://github.com/nodeca/js-yaml/issues/345#issuecomment-734812253

Here is a lockfile exhibiting the problem (look for `@nodelib/fs.scandir@npm:2.1.3`, which is present twice):
https://gist.github.com/arcanis/3778f61718474f0e647f847986895077

**Comment by bertho-zero** [comment]
Can you publish a version of `@yarnpkg/parsers` with this fix? I also encounter this error.

**Comment by merceyz** [comment]
@bertho-zero It was released in `@yarnpkg/parsers@2.3.1-rc.1`
https://github.com/yarnpkg/berry/releases/tag/%40yarnpkg%2Fparsers%2F2.3.1-rc.1
